test/junit/src/test/scala/strawman/collection/immutable/SortedMapTest.scala

```diff
@@ -0,0 +1,94 @@
+package strawman.collection.immutable
+
+import org.junit.Assert.{assertEquals, assertNotEquals}
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(classOf[JUnit4])
+class SortedMapTest {
+  @Test
+  def testWithDefaultValueReturnsSortedMapWithDefaultValue(): Unit = {
+    val tree: SortedMap[Int, String] = SortedMap.from(Map(1 -> "One", 2 -> "Two"))
+      .withDefault(defaultValueFunction)
+
+    assertEquals("3 is not present in this map", tree(3))
+  }
+
+  @Test
+  def testDefaultValueIsPersistedAfterAnElementIsAddedToUnderlyingMap(): Unit = {
+    val tree: SortedMap[Int, String] = SortedMap.from(Map(1 -> "One", 2 -> "Two")).withDefault(defaultValueFunction)
+    val newTree = tree + (3 -> "Three")
+
+    assertEquals("5 is not present in this map", newTree(5))
+  }
+
+  @Test
+  def testDefaultValueIsPersistedAfterCreatingEmptyMapFromUnderlyingSortedMap(): Unit = {
+    val emptyMap: SortedMap[Int, String] = SortedMap.from(Map(1 -> "One", 2 -> "Two"))
+      .withDefault(defaultValueFunction)
+      .empty
+
+    assertEquals("1 is not present in this map", emptyMap(1))
+  }
+
+  @Test
+  def testDefaultValueIsPersistedAfterRangeOperationOnSortedMap(): Unit = {
+    val rangedProjectOfTreeMap: SortedMap[Int, String] = SortedMap.from(Map(1 -> "One", 2 -> "Two", 3 -> "Three"))
+      .withDefault(defaultValueFunction)
+      .range(1, 3) // range `to` parameter is not inclusive
+
+    assertEquals("3 is not present in this map", rangedProjectOfTreeMap(3))
+  }
+
+  @Test
+  def testDefaultValueIsPersistedAfterAnElementFromUnderlyingMapIsRemoved(): Unit = {
+    val originalTreeMap : SortedMap[Int, String] = SortedMap.from(Map(1 -> "One", 2 -> "Two", 3 -> "Three"))
+      .withDefault(defaultValueFunction)
+
+    val newTreeMap = originalTreeMap - 3
+
+    assertEquals("3 is not present in this map", newTreeMap(3))
+    assertEquals(2, newTreeMap.size)
+  }
+
+  @Test
+  def testDefaultValueIsLostWhenNewSortedMapIsCreatedFromIterablesInOperationsLikeFlatMap(): Unit = {
```

Comment:
I’m wondering whether we should take advantage of the collections rewrite to fix that limitation (and to fix it in `Map.WithDefault` as well).

Do you think people rely on the fact that transformation operations such as `map` and `flatMap` do not preserve the default value? (cc @lrytz and @szeiger)

Reply:
That's a good idea. But if we have to persist this value then we will have to put constraints on type variance of `K, V` parameters when someone uses `withDefault` operations. This will be more constraining than the compromise of losing a default value when underlying types of Iterables are changed due to operations such as `map` and `flatmap`

Reply:
Oh, good point, I missed that…